sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PropertiesShouldBePreferred.cs

```diff
@@ -68,31 +71,31 @@ protected override void Initialize(SonarAnalysisContext context)
         }
 
         private bool IsPropertyCanditate(IMethodSymbol method)
-        {
-            if (method.IsConstructor() ||
-                method.MethodKind == MethodKind.PropertyGet ||
-                method.IsOverride ||
-                method.GetInterfaceMember() != null ||
-                method.IsAsync ||
-                method.ReturnType.OriginalDefinition.IsAny(KnownType.SystemTasks))
-            {
-                return false;
-            }
-
-            return method.Parameters.Length == 0 &&
-                   !method.ReturnsVoid &&
-                   !method.ReturnType.Is(TypeKind.Array) &&
-                   method.Name != "GetEnumerator" &&
-                   method.Name != "GetAwaiter" &&
-                   NameStartsWithGet(method);
-        }
+            => method.Parameters.Length == 0
+            && method.GetInterfaceMember() is null
+            && !method.IsConstructor()
+            && !method.IsOverride
+            && !method.ReturnsVoid
+            && !method.ReturnType.Is(TypeKind.Array)
+            && !method.IsAsync
+            && !method.ReturnType.OriginalDefinition.IsAny(KnownType.SystemTasks)
+            && method.MethodKind != MethodKind.PropertyGet
+            && method.Name != "GetEnumerator"
+            && method.Name != "GetAwaiter";
 
-        private bool NameStartsWithGet(IMethodSymbol method)
+        private static bool NameStartsWithGet(IMethodSymbol method)
         {
             var nameParts = method.Name.SplitCamelCaseToWords().ToList();
-
             return nameParts.Count > 1 &&
                    nameParts[0] == "GET";
         }
+
+        private static bool AllAttributesAllowProperties(IMethodSymbol method)
+            => method.GetAttributes()
+            .Select(attribute => attribute.AttributeClass)
+            .SelectMany(cls => cls.GetAttributes())
+            .Where(attr => attr.AttributeClass.Is(KnownType.System_AttributeUsageAttribute))
+            .Select(attr => attr.ConstructorArguments[0].Value).Cast<AttributeTargets>()
```

Comment:
To be on the safe side
```suggestion
            .Select(attr => attr.ConstructorArguments[0].Value)
            .OfType<AttributeTargets>()
```